src/libraries/System.Net.Primitives/src/System/Net/CookieContainer.cs

```diff
@@ -1057,7 +1058,7 @@ internal ICollection Values
             }
         }
 
-        internal object this[string s]
+        internal object? this[string s]
```

Comment:
I think this should be non-nullable

Reply:
From https://docs.microsoft.com/en-us/dotnet/api/system.collections.sortedlist.item?view=netframework-4.8
> A key cannot be null, but a value can be.

Reply:
Just because the underlying collection being used supports null values doesn't mean null values are supported here.  As far as I can tell you can't add anything null to the collection, so null will never be returned.

Reply:
But I can specify a key that does not exist. In the case we should either throw or return null.

Reply:
It would throw rather than return null.

Reply:
There is  a code:
```c#
                lock (pathList.SyncRoot)
                {
                    cookies = (CookieCollection?)pathList[cookie.Path]!;

                    if (cookies == null)
                    {
                        cookies = new CookieCollection();
                        pathList[cookie.Path] = cookies;
                    }
                }
```
that means the indexer could return null.

Reply:
Could it just be that the check is too defensive? Do we have evidence that a null value can be set either in the implementation or via the public api? I don't have a strong opinion here, just asking so we make right decision.

Reply:
The current internal implementation doesn't pass nulls here. But the implementation would be changed later.  So because it is _internal_ indexer I believe it is dangerous to remove `?`.